java/connector-node/risingwave-sink-dynamic/risingwave-sink-dynamic-service/src/main/java/com/risingwave/connector/context/MailBoxExecImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.risingwave.connector.context;
+
+import java.util.Arrays;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.apache.flink.api.common.operators.MailboxExecutor;
+import org.apache.flink.util.FlinkRuntimeException;
+import org.apache.flink.util.function.ThrowingRunnable;
+
+/**
+ * Simple Implementation of Flink MailboxExecutor. Only the mailbox method of executing a mail via
+ * `yeild` call is implemented. In flink, there is also a background thread that polls for
+ * execution, which is not implemented because it is not currently used.
+ *
+ * <p>Specifically, the class has a queue of different mails saved in the queue, and passed
+ * downstream, when the downstream call yeild, back to the fifo way to pick the mail execution
+ */
+public class MailBoxExecImpl implements MailboxExecutor {
+    BlockingQueue<Mail> queue = new LinkedBlockingQueue<>();
```

Comment:
May add a comment that the created queue is of Integer.MAX capacity and will not be bounded.